nixos/modules/services/video/mirakurun.nix

```diff
@@ -0,0 +1,131 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mirakurun;
+  mirakurun = pkgs.mirakurun;
+
+  serverConfig = recursiveUpdate cfg.extraConfig {
+    inherit (cfg) logLevel port;
+    path = cfg.domainSocket;
+  };
+
+in
+  {
+    options = {
+      services.mirakurun = {
+        enable = mkEnableOption mirakurun.meta.description;
+
+        group = mkOption {
+          type = types.str;
+          default = "video";
+          description = "Group which mirakurun belongs to.";
+        };
+
+        logLevel = mkOption {
+          type = types.ints.between (-1) 3;
+          default = 2;
+          description = "Mirakurun log level.";
+        };
+
+        domainSocket = mkOption {
+          type = with types; nullOr path;
+          default = "/var/run/mirakurun/mirakurun.sock";
+          description = ''
+            Path to domain socket used to communicate with Mirakurun.
+          '';
+        };
+
+        port = mkOption {
+          type = with types; nullOr port;
+          default = 40772;
+          description = "Port to listen on.";
+        };
+
+        openFirewall = mkOption {
+          type = types.bool;
+          default = false;
+          description = ''
+            Open ports in the firewall for the Mirakurun web interface.
+          '';
+        };
+
+        extraConfig = mkOption {
+          type = types.attrs;
```

Comment:
https://github.com/NixOS/nixpkgs/pull/75584 is now merged, see https://hydra.nixos.org/build/124706541/download/2/nixos/index.html#sec-settings-options how you can use it to improve this module

Reply:
Cool! I've replaced `extraConfig` with `serverSettings` to make use of this feature.